tests/test_cli.py

```diff
@@ -315,6 +330,40 @@ def test_run_with_user_yamllint_config_file_in_env(self):
                 cli.run((os.path.join(self.wd, 'a.yaml'), ))
             self.assertEqual(ctx.returncode, 1)
 
+    def test_run_with_locale(self):
+        self.addCleanup(locale.setlocale, locale.LC_ALL, (None, None))
+        try:
+            locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
+        except locale.Error:
+            self.skipTest('locale en_US.UTF-8 not available')
+        locale.setlocale(locale.LC_ALL, (None, None))
```

Comment:
Can you briefly explain why you added this? Is the cleanup on line 334 redundant?

Reply:
The cleanup is still needed after the whole test ran.

Setting the locale to `en_US` is just a test for locale availability. But because the default without locale option (as in the first of the for test `.run`s below) doesn't set any locale, that `en_US` would still be active. Therefore we need to reset that to it's initial state first, then run the 2 tests with default settings, then the 2 with locale setting, then cleanup.

Reply:
Thanks!

I understood the second one (line 339), but I thought the cleanup wasn't needed, then. Actually it's still needed because yamllint calls `locale.setlocale()` too (in `cli.py`).

Could you add a tiny comment above `self.addCleanup()`, just to make sure we remember it? E.g. `# Clean up after yamllint called locale.setlocale() in cli.py:`

Reply:
I added some comments and moved the `addCleanup` to a more appropriate spot regarding the flow in this test.